plugins/main/public/components/common/doc_viewer/doc_viewer.tsx

```diff
@@ -0,0 +1,150 @@
+
+const COLLAPSE_LINE_LENGTH = 350;
+const DOT_PREFIX_RE = /(.).+?\./g;
+
+export type tDocViewerProps = {
+    flattened: any;
+    formatted: any;
+    mapping: any;
+    indexPattern: any;
+}
+
+/**
+ * Convert a dot.notated.string into a short
+ * version (d.n.string)
+ */
+export const shortenDottedString = (input: string) => input.replace(DOT_PREFIX_RE, '$1.');
+
+export function getFieldTypeName(type: string) {
+    switch (type) {
+        case 'boolean':
+            return i18n.translate('discover.fieldNameIcons.booleanAriaLabel', {
+                defaultMessage: 'Boolean field',
+            });
+        case 'conflict':
+            return i18n.translate('discover.fieldNameIcons.conflictFieldAriaLabel', {
+                defaultMessage: 'Conflicting field',
+            });
+        case 'date':
+            return i18n.translate('discover.fieldNameIcons.dateFieldAriaLabel', {
+                defaultMessage: 'Date field',
+            });
+        case 'geo_point':
+            return i18n.translate('discover.fieldNameIcons.geoPointFieldAriaLabel', {
+                defaultMessage: 'Geo point field',
+            });
+        case 'geo_shape':
+            return i18n.translate('discover.fieldNameIcons.geoShapeFieldAriaLabel', {
+                defaultMessage: 'Geo shape field',
+            });
+        case 'ip':
+            return i18n.translate('discover.fieldNameIcons.ipAddressFieldAriaLabel', {
+                defaultMessage: 'IP address field',
+            });
+        case 'murmur3':
+            return i18n.translate('discover.fieldNameIcons.murmur3FieldAriaLabel', {
+                defaultMessage: 'Murmur3 field',
+            });
+        case 'number':
+            return i18n.translate('discover.fieldNameIcons.numberFieldAriaLabel', {
+                defaultMessage: 'Number field',
+            });
+        case 'source':
+            // Note that this type is currently not provided, type for _source is undefined
+            return i18n.translate('discover.fieldNameIcons.sourceFieldAriaLabel', {
+                defaultMessage: 'Source field',
+            });
+        case 'string':
+            return i18n.translate('discover.fieldNameIcons.stringFieldAriaLabel', {
+                defaultMessage: 'String field',
+            });
+        case 'nested':
+            return i18n.translate('discover.fieldNameIcons.nestedFieldAriaLabel', {
+                defaultMessage: 'Nested field',
+            });
+        default:
+            return i18n.translate('discover.fieldNameIcons.unknownFieldAriaLabel', {
+                defaultMessage: 'Unknown field',
+            });
+    }
+}
+
+const DocViewer = (props: tDocViewerProps) => {
+    const [fieldRowOpen, setFieldRowOpen] = useState({} as Record<string, boolean>);
+    const { flattened, formatted, mapping, indexPattern } = props;
+
+    return (<>
+        {flattened && (
+            <table className="table table-condensed osdDocViewerTable">
```

Comment:
Why don't you use Elastic's React Table component?

Reply:
I'd tried to keep the component without dependencies using HTML. I'd removed the class names because are not used.